test/sage_test.exs

```diff
@@ -254,6 +254,70 @@ defmodule SageTest do
     end
   end
 
+  describe "interleaves/3" do
+    test "adds a step between every transaction" do
+      sage =
+        new()
+        |> run(:t1, transaction(:t1))
+        |> run(:t2, transaction(:t2))
+        |> run_async(:t_async, transaction(:t_async), :noop)
+        |> run(:t3, transaction(:t3))
+        |> interleave(:i, fn _effects, _args, previous_stage_name -> {:ok, previous_stage_name} end)
+
+      assert [i_4: _, t3: _, i_3: _, t_async: _, i_2: _, t2: _, i_1: _, t1: _] = sage.stages
+      assert {:ok, _, %{i_4: :t3, i_3: :t_async, i_2: :t2, i_1: :t1}} = execute(sage)
+    end
+
+    test "adds nothing if there are no transactions" do
+      assert [] =
+               new()
+               |> interleave(:i, fn _effects, _args, _previous_stage_name -> :ok end)
+               |> Map.get(:stages)
```

Comment:
Can we change the code style a bit here? Something like
```suggestion
      sage = 
        new() 
        |> interleave(:i, fn _effects, _args, _previous_stage_name -> :ok end)
     
      assert sage.stages == []
```

This is about the same number of lines, but formatting looks a bit more readable. Plus we want literal values to be on the right and only match values on the left. Such code style allow not to make a mistake and match when we actually wanted a comparison (eg. a typical beginners mistake is `assert %{} = empty_map`).